### PR TITLE
GEOMESA-786 STIdxStrategy improperly sets STII filter when whole world

### PR DESCRIPTION
**Pull request by AndrewAnnex**
* Added test for if filter is whole world in STIdxStrategy

Signed-off-by: Andrew Annex <aannex@ccri.com>

### CODE REVIEW

File: geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala [hunk 1]
```diff
@@ -162,7 +171,10 @@ class STIdxStrategy extends Strategy with Logging with IndexFilterHelpers {
     val cfg = new IteratorSetting(iteratorPriority_SpatioTemporalIterator,
       "within-" + randomPrintableString(5),classOf[IndexIterator])
 
-    configureStFilter(cfg, filter)
+    // test if the filter is for the whole world
+    val isWholeWorld = filter.exists(isFilterWholeWorld)
+    if (!isWholeWorld) configureStFilter(cfg, filter)
```
Comment (anthonyccri) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala]: Does this keep the fine resolution time filter?

File: geomesa-core/src/test/scala/org/locationtech/geomesa/core/data/AccumuloDataStoreTest.scala [hunk 2]
```diff
@@ -1155,6 +1154,82 @@ class AccumuloDataStoreTest extends Specification {
...
+      val explain = {
+        val o = new ExplainString
+        ds.explainQuery(sftName, query, o)
+        o.toString()
+      }
+      ds.removeSchema(sftName)
+      explain must not contain("STII Filter: [ geom bbox ")
+      explain must contain("No STII Filter")
+      explain must contain("Filter: AcceptEverythingFilter")
+    }.pendingUntilFixed("Fixed query planner to deal with OR'd whole world geometry")
```
Comment (jnh5y) [inline geomesa-core/src/test/scala/org/locationtech/geomesa/core/data/AccumuloDataStoreTest.scala]: @AndrewAnnex toss a link to the JIRA ticket in here.

File: geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala [hunk 3]
```diff
@@ -87,11 +82,16 @@ class STIdxStrategy extends Strategy with Logging with IndexFilterHelpers {
     val temporal = extractTemporal(dtgField)(temporalFilters)
     val interval = netInterval(temporal)
     val geometryToCover = netGeom(collectionToCover)
-    val filter = buildFilter(geometryToCover, interval)
 
-    output(s"GeomsToCover: $geomsToCover")
+    //This is correctly doing an AcceptEverythingFilter if the geometryToCover is the whole world
+    val filter = buildFilter(geometryToCover, interval)
+    // This catches the case when a whole world query slips through DNF/CNF
+    // The union on this geometry collection is necessary at the moment but is not true
```
Comment (jnh5y) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala]: Hmmm?  This sentence almost makes.

File: geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala [hunk 4]
```diff
@@ -87,11 +82,16 @@ class STIdxStrategy extends Strategy with Logging with IndexFilterHelpers {
...
+    // If given spatial predicates like disjoint.
+    val ofilter = if (geometryToCover != null && isWholeWorld(geometryToCover.union)) {
```
Comment (jnh5y) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala]: I'd suggest adding a null check to isWholeWorld so that this call can be simplified.

File: geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala (hunk 1 above)
Comment (jnh5y) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala]: @anthonyccri @AndrewAnnex This is a great question.  

tl;dr: I think the check is unnecessary and should be removed.

Check out this: https://github.com/jnh5y/geomesa/commit/9ca8d2657b0d10a534cb19ebefe017a3db29ca1c

If there is a whole world clause, it has been removed by this point.

File: geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala (hunk 4 above)
Comment (AndrewAnnex) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala]: "isWholeWorld" returns a boolean. Could you clarify please?

Comment (jnh5y) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala]: Claim 1: isWholeWorld(null) returns false.  
Claim 2: isWholeWorld(gc: GeometryCollection) = isWholeWorld(gc.union).

If those claims make sense, updating isWholeWorld to handle those cases makes it more robust.  Additionally, this line becomes, 
....if(isWholeWorld(geometryToCover)...

**Comment by jnh5y** [comment]
Merged as ff4405ad84a52a65febbc3a423c7d01b8e5ddbae on 1.5.
